lib/middleware/middleware.go

```diff
@@ -0,0 +1,75 @@
+package middleware
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib/utils"
+)
+
+// BuildRundeckURLEndpoint builds a URL endpoint given a baseURL and an endpoint.
+// It returns the full URL endpoint and an error if any.
+func BuildRundeckURLEndpoint(baseURL, endpoint string) (string, error) {
+	base, err := url.Parse(baseURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse Rundeck base URL")
+	}
+
+	queryEndpoint, err := url.Parse(endpoint)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse Rundeck query Endpoint")
```

Comment:
What do you think about including `queryEndpoint` in the error message?

Reply:
The queryEndpoint includes the API auth token in clear text. Maybe we can mask it as we return the response?